controller/aks-cluster-config-handler.go

```diff
@@ -633,9 +633,9 @@ func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.Secr
 	}
 
 	// set addon monitoring profile
-	if addonProfile["omsagent"] != nil {
-		upstreamSpec.Monitoring = addonProfile["omsagent"].Enabled
-		logAnalyticsWorkspaceResourceID := addonProfile["omsagent"].Config["logAnalyticsWorkspaceResourceID"]
+	if addonProfile["omsAgent"] != nil {
```

Comment:
nit: maybe just put a var here for omsAgent and use it for these three instances.